cartesian_controllers/test/test_twist_controller.cpp

```diff
@@ -149,20 +151,15 @@ TEST_F(TwistControllerTest, command_callback_test)
     ASSERT_EQ(command_itfs_[4].get_value(), 0.0);
     ASSERT_EQ(command_itfs_[5].get_value(), 0.0);
 
-    auto node_state = controller_->configure();
+    auto node_state = controller_->get_node()->configure();
     ASSERT_EQ(node_state.id(), lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE);
 
     node_state = controller_->get_node()->activate();
     ASSERT_EQ(node_state.id(), lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
 
-    // send a new command
+    // send a new command and wait
     publish_commands();
-
-    // wait for command message to be passed
-    ASSERT_EQ(wait_for(controller_->twist_command_subscriber_), rclcpp::WaitResultKind::Ready);
-
-    // process callbacks
-    rclcpp::spin_some(controller_->get_node()->get_node_base_interface());
+    ASSERT_TRUE(controller_->wait_for_commands(executor));
```

Comment:
This `wait_for` resulted in rclcpp throwing an exception, there was a `wait_for_commands` which looks like it was intended for this. A similar pattern exists in the other controller tests.
https://github.com/ros-controls/ros2_controllers/blob/6793e0278f35b34800c14a085188f3d7a3b4f6ad/admittance_controller/test/test_admittance_controller.cpp#L268-L269